### PR TITLE
Convert ASM to mixin

### PR DESCRIPTION
**Pull request by Lyfts**
Converts all the asm to mixins and ditches mysteriumlib. There's likely some bugs or something I missed so I'd love it if someone wanted to test it out.

A couple of the transformers were replacing inventory array calls with `BackhandUtils.setPlayerCurrentItem`, this is an inherited thing from bg2 and from my testing doesn't seem to be required for Backhand and have thus been removed & not converted.

### CODE REVIEW

**Comment by Caedis** [comment]
Once this is merged, I am going to work on a way to recover any items in the BG2 slots.

**Comment by Caedis** [comment]
Slowly going through this as I also use it in playthrough

File: src/main/java/xonin/backhand/mixins/early/MixinNetHandlerPlayServer.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    @Shadow
+    public EntityPlayerMP playerEntity;
+
+    @ModifyExpressionValue(
+        method = "processHeldItemChange",
+        at = @At(
+            value = "INVOKE",
+            target = "Lnet/minecraft/network/play/client/C09PacketHeldItemChange;func_149614_c()I",
+            ordinal = 1))
+    private int backhand$isValidIventorySlot(int original) {
```
Comment (Caedis) [inline src/main/java/xonin/backhand/mixins/early/MixinNetHandlerPlayServer.java:33]: nit: Iventory -> Inventory

**Review by Caedis — APPROVED** [review]

File: src/main/java/xonin/backhand/mixins/early/MixinNetHandlerPlayServer.java (hunk 1 above)
Comment (Lyfts) [reply src/main/java/xonin/backhand/mixins/early/MixinNetHandlerPlayServer.java:33]: https://github.com/GTNewHorizons/Battlegear2/blob/3f4c858146ecf588664ced4d2b158c090d66fd1d/src/main/java/mods/battlegear2/mixins/early/MixinNetHandlerPlayServer.java#L31
BG2 did me dirty

**Review by Caedis — APPROVED** [review]

**Pull request merged**